### PR TITLE
Adds testing to basic function and fixing found edge cases

### PR DESCRIPTION
**Pull request by CollinLFerguson**
Adds some simple tests to the UI Library, fixes some breaking edge cases, mostly around nulls or type checking.

### CODE REVIEW

File: src/lib/components/containers/charts/ScatterplotChart.svelte
Comment (jesseinc92) [inline src/lib/components/containers/charts/ScatterplotChart.svelte:1]: Remove this console log since we don't want this running in production.

File: src/lib/functions/formatting/convertToCurrency.js
Comment (jesseinc92) [inline src/lib/functions/formatting/convertToCurrency.js:1]: Fix misspelling "valudated"

File: src/lib/functions/formatting/parseURL.js
Comment (jesseinc92) [inline src/lib/functions/formatting/parseURL.js:1]: Add console.warn statement to bring functionality in line with the rest of the changes you've made on other utilities.

File: tests/unit/functions/charts/consolodateMonths.test.js
Comment (jesseinc92) [inline tests/unit/functions/charts/consolodateMonths.test.js:1]: Remove incorrect consolidateMonths import. Fix misspellings "consolodate" throughout.

File: tests/unit/functions/charts/consolodateMonths.test.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        let testDate3 = new Date("2023-01-01T05:00:00.000Z")
+        let consolodatedMonths3 = consolidateMonths(chartData3, "date", "value")
+        
+        // console.log(consolodatedMonths3)
+    })
+})
+
+describe(`Tests ${functionName} for additional special cases:`, ()=>{
+	it(`Tests ${functionName} for additional special case: [case]:`, ()=>{ //repeat as need.
+		true
```
Comment (jesseinc92) [inline tests/unit/functions/charts/consolodateMonths.test.js:151]: Is there meant to be a test condition here?

File: tests/unit/functions/charts/consolodateYears.test.js
Comment (jesseinc92) [inline tests/unit/functions/charts/consolodateYears.test.js:1]: Remove incorrect import. Fix any misspellings of "consolodated".

File: tests/unit/functions/charts/consolodateYears.test.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            ]
+        
+        let chartData3 = JSON.parse(JSON.stringify(data3))
+        for (let obj of chartData3) {
+            obj[domain] = new Date(obj[domain])
+        }
+        let testDate3 = new Date("2023-01-01T05:00:00.000Z")
+        let consolodatedMonths3 = consolodateYears(chartData3, "date", "value")
+        
+        //console.log(consolodatedMonths3)
```
Comment (jesseinc92) [inline tests/unit/functions/charts/consolodateYears.test.js:141]: Is there meant to be a test condition here?

File: tests/unit/functions/filtering/withinDateRange.test.js
Comment (jesseinc92) [inline tests/unit/functions/filtering/withinDateRange.test.js:1]: Remove console log. Otherwise good!

File: tests/unit/functions/filtering/withinTimeRange.test.js
Comment (jesseinc92) [inline tests/unit/functions/filtering/withinTimeRange.test.js:1]: Clean up indent formatting – otherwise good!

File: src/lib/functions/formatting/convertToCurrency.js
Comment (CollinLFerguson) [reply src/lib/functions/formatting/convertToCurrency.js:1]: Great catch. resolved

File: tests/unit/functions/filtering/withinDateRange.test.js
Comment (CollinLFerguson) [reply tests/unit/functions/filtering/withinDateRange.test.js:1]: Good catch, resolved.
